src/hooks/useGridParticipants/useGridParticipants.ts

```diff
@@ -0,0 +1,94 @@
+import { RemoteParticipant } from 'twilio-video';
+import { useEffect, useState } from 'react';
+import useDominantSpeaker from '../useDominantSpeaker/useDominantSpeaker';
+import { useAppState } from '../../state';
+import useVideoContext from '../useVideoContext/useVideoContext';
+
+// If an offscreen participant becomes the dominant speaker, we
+// move them onscreen to where the least recent onscreen dominant
+// speaker was located. We are able to order the speakers appropriately
+// by keeping track of the timestamp from when they became the newest
+// dominant speaker.
+
+interface OrderedParticipant {
+  participant: RemoteParticipant;
+  dominantSpeakerStartTime: number;
+}
+
+export default function useGridParticipants() {
+  const [orderedParticipants, setOrderedParticipants] = useState<OrderedParticipant[]>([]);
+  const { room } = useVideoContext();
+  const dominantSpeaker = useDominantSpeaker(true);
```

Comment:
Do we want to `includeNull` here?

Reply:
Nope, looks like it isn't needed. I thought it was so that when the dominant speaker stops talking (and becomes null), they wouldn't be highlighted in green any more. Now I realize that by passing `includeNull` in the `<GridView />` component, this is taken care of.